routes/topics.js

```diff
@@ -16,6 +16,8 @@ router.get('/follow', isAuth, topicsController.getFollowTopic);
 router.get('/latest', validate(TopicValidation.latestPost), isAuth, topicsController.getLatestPost);
 router.put('/follow', isAuth, topicsController.putFollowTopic);
 router.put('/follow-v2', isAuth, topicsController.followTopicV2);
+router.get('/is-exist', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
+router.post('/create', validate(TopicValidation.create), isAuth, topicsController.create);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new routes `/is-exist` and `/create` are added correctly with appropriate validation and authentication middleware. However, the route naming could be improved for better readability and RESTful API design principles. Instead of using `/is-exist`, consider using a more conventional route like `/exists`. Similarly, for creating a new topic, you can use a POST request on the base route `/` instead of `/create`.
```diff
- router.get('/is-exist', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
- router.post('/create', validate(TopicValidation.create), isAuth, topicsController.create);
+ router.get('/exists', validate(TopicValidation.checkName), isAuth, topicsController.checkName);
+ router.post('/', validate(TopicValidation.create), isAuth, topicsController.create);
```
This way, your API endpoints will be more intuitive and adhere to standard conventions.


<!-- This is an auto-generated comment by OSS CodeRabbit -->